plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -4488,6 +4483,34 @@ public void testOptimizeCleansUpDeleteFiles()
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    @Test
+    public void testOptimizeSnapshot()
+    {
+        Session session = Session.builder(getSession())
+                .setCatalogSessionProperty(getSession().getCatalog().orElseThrow(), "allow_legacy_snapshot_syntax", "true")
+                .build();
+        String tableName = "test_optimize_snapshot_" + randomTableSuffix();
+
+        assertUpdate("CREATE TABLE " + tableName + " (a) AS VALUES 11", 1);
+        long snapshotId = getCurrentSnapshotId(tableName);
+        assertUpdate("INSERT INTO " + tableName + " VALUES 22", 1);
+        assertThatThrownBy(() -> query(session, "ALTER TABLE \"%s@%d\" EXECUTE OPTIMIZE".formatted(tableName, snapshotId)))
+                .hasMessage("Cannot execute table procedure OPTIMIZE on old snapshot " + snapshotId);
+        assertThat(query("SELECT * FROM " + tableName))
+                .matches("VALUES 11, 22");
```

Comment:
nit: No change requested. `assertQuery` looks better for simple query assertion.

Reply:
agree. i think i perhaps did this more for consistency reasons.